ledger/participant-integration-api/src/main/scala/platform/store/cache/MutableCacheBackedContractStore.scala

```diff
@@ -302,51 +301,29 @@ object MutableCacheBackedContractStore {
       ContractsStateCache(maxContractsCacheSize, metrics),
     )
 
-  def owner(
-      contractsReader: LedgerDaoContractsReader,
-      signalNewLedgerHead: SignalNewLedgerHead,
-      metrics: Metrics,
-      maxContractsCacheSize: Long,
-      maxKeyCacheSize: Long,
-  )(implicit
-      executionContext: ExecutionContext,
-      loggingContext: LoggingContext,
-  ): Resource[MutableCacheBackedContractStore] =
-    Resource.successful(
-      MutableCacheBackedContractStore(
-        contractsReader,
-        signalNewLedgerHead,
-        metrics,
-        maxContractsCacheSize,
-        maxKeyCacheSize,
-      )
-    )
-
   def ownerWithSubscription(
       subscribeToContractStateEvents: SubscribeToContractStateEvents,
       contractsReader: LedgerDaoContractsReader,
       signalNewLedgerHead: SignalNewLedgerHead,
       metrics: Metrics,
       maxContractsCacheSize: Long,
       maxKeyCacheSize: Long,
+      executionContext: ExecutionContext,
       minBackoffStreamRestart: FiniteDuration = 100.millis,
   )(implicit
       materializer: Materializer,
       loggingContext: LoggingContext,
-      executionContext: ExecutionContext,
-      resourceContext: ResourceContext,
-  ): Resource[MutableCacheBackedContractStore] = {
-
-    val contractStoreOwner = owner(
-      contractsReader = contractsReader,
-      signalNewLedgerHead = signalNewLedgerHead,
-      metrics = metrics,
-      maxContractsCacheSize = maxContractsCacheSize,
-      maxKeyCacheSize = maxKeyCacheSize,
-    )
+  ): ResourceOwner[MutableCacheBackedContractStore] = {
+    val contractStore = MutableCacheBackedContractStore(
+      contractsReader,
+      signalNewLedgerHead,
+      metrics,
+      maxContractsCacheSize,
+      maxKeyCacheSize,
+    )(executionContext, loggingContext)
 
-    val subscribingContractStoreOwner = (contractStore: MutableCacheBackedContractStore) =>
-      ResourceOwner.forCloseable[AutoCloseable](() =>
+    ResourceOwner
+      .forCloseable(() =>
         new AutoCloseable {
```

Comment:
This whole construct seems a bit suspicious. I suggest making this whole thing a `class OwnerWithSubscription extends ResourceOwner[MutableCachedBackContractStore] { … }` and returning a `Resource(open)(close)`.

Declaring the `contractStore` should probably be in the `acquire` method too, not in the resource owner constructor. Depends if it's OK for acquiring twice to share a reference.

Reply:
Good suggestion. Applied it